examples/win-offsets.c

```diff
@@ -0,0 +1,131 @@
+/* The LibVMI Library is an introspection library that simplifies access to
+ * memory in a target virtual machine or in a file containing a dump of
+ * a system's physical memory.  LibVMI is based on the XenAccess Library.
+ *
+ * Author: Sergey Kovalev (valor@list.ru)
+ *
+ * This file is part of LibVMI.
+ *
+ * LibVMI is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * LibVMI is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with LibVMI.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#define _GNU_SOURCE
+
+#include <libvmi/libvmi.h>
+#include <libvmi/peparse.h>
+#include <libvmi/os/windows/windows.h>
```

Comment:
This is a private header of LibVMI. You should not include it in an example.

Reply:
I need it to print internal offsets. Then I could use it to create _libvmi.conf_.

Reply:
The internal offsets are created _from_ the configuration that is passed _in_ to LibVMI. If you want a tool that automatically creates libvmi.conf take a look at `tools/config-editor`, `tools/linux-offset-finder` and `tools/windows-offset-finder`

Reply:
Unfortunately I didn't find a way to get _ntoskrnl_ and _ntoskrnl_va_. Both are used in path of `windows_init()` if I understand the code. So the alternative is to add getters for it in `libvmi/accessors.c`

Reply:
Sure, you can add new accessor functions to libvmi.h, there are is no explicit `vmi_get_kernel_base` function at the moment. I personally found getting a kernel symbol enough as based on that you can calculate the kernel base yourself. But having dedicated functions for this would probably.be more straight forward.

Reply:
I have decided to extend `windows_get_offset()` to return all essential fields. Is it good enough?